File: ios/objc/StorylyDemo/StorylyDemo/Basic/BasicViewController.m
//
//  ViewController.m
//  StorylyDemo
//
//  Created by Levent ORAL on 18.02.2020.
//  Copyright © 2020 App Samurai Inc. All rights reserved.
//

#import "BasicViewController.h"
#import "Tokens.h"

@interface BasicViewController ()
@property (weak, nonatomic) IBOutlet StorylyView *storylyView;
@end

@implementation BasicViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.storylyView.delegate = self;
    self.storylyView.rootViewController = self;
    StorylyView *storylyViewProgrammatic = [[StorylyView alloc] init];
    storylyViewProgrammatic.rootViewController = self;
    storylyViewProgrammatic.translatesAutoresizingMaskIntoConstraints = NO;
    [self.view addSubview:storylyViewProgrammatic];
    
    [[storylyViewProgrammatic.leadingAnchor constraintEqualToAnchor:self.view.leadingAnchor] setActive:YES];
    [[storylyViewProgrammatic.trailingAnchor constraintEqualToAnchor:self.view.trailingAnchor] setActive:YES];
    [[storylyViewProgrammatic.topAnchor constraintEqualToAnchor:self.storylyView.bottomAnchor constant:50] setActive:YES];
    [[storylyViewProgrammatic.heightAnchor constraintEqualToConstant:120] setActive:YES];
    

    StorylyStoryGroupStyling *groupStyling = [[[[[[[StorylyStoryGroupStylingBuilder alloc] init]
                                              setSize:StoryGroupSizeCustom]
                                              setIconHeight:120]
                                              setIconWidth:120]
                                              setIconCornerRadius:12]
                                              build];
       
    StorylyBarStyling *barStyling = [[[[StorylyBarStylingBuilder alloc] init]
                                     setHorizontalPaddingBetweenItems:15]
                                     build];
    
    StorylyStoryStyling *storyStyling = [[[[StorylyStoryStylingBuilder alloc] init]
                                         setTitleColor: UIColor.redColor]
                                         build];
             
    StorylyConfig *storylyConfig = [[[[[[StorylyConfigBuilder alloc] init]
                                     setStoryGroupStyling:groupStyling]
                                     setBarStyling:barStyling]
                                     setStoryStyling:storyStyling]
                                     build];
         
       
    storylyViewProgrammatic.storylyInit = [[StorylyInit alloc] initWithStorylyId:STORYLY_INSTANCE_TOKEN config:storylyConfig];
    self.storylyView.storylyInit = [[StorylyInit alloc] initWithStorylyId:STORYLY_INSTANCE_TOKEN config:storylyConfig];
}

- (void)storylyLoaded:(StorylyView *)storylyView
       storyGroupList:(NSArray<StoryGroup *> *)storyGroupList
           dataSource:(enum StorylyDataSource)dataSource {
    NSLog(@"[storyly] IntegrationViewController:storylyLoaded - storyGroupList size {%lu)} - source {%ld)}", (unsigned long)storyGroupList.count, (long)dataSource);
}

- (void)storylyLoadFailed:(StorylyView *)storylyView
             errorMessage:(NSString *)errorMessage {
    NSLog(@"[storyly] IntegrationViewController:storylyLoadFailed - error {%@)}", errorMessage);
}

- (void)storylyActionClicked:(StorylyView *)storylyView
          rootViewController:(UIViewController *)rootViewController
                       story:(Story *)story {
    NSLog(@"[storyly] IntegrationViewController:storylyActionClicked - story action_url {%@)}", story.media.actionUrl);
    [storylyView closeStoryWithAnimated:true completion:nil];
}

@end

File: ios/swift/StorylyDemo/StorylyDemo/Basic/BasicViewController.swift
//
//  BasicViewController.swift
//  StorylyDemo
//
//  Created by Levent ORAL on 25.09.2019.
//  Copyright © 2019 App Samurai Inc. All rights reserved.
//

import UIKit
import Storyly

class BasicViewController: UIViewController {

    @IBOutlet weak var storylyView: StorylyView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.storylyView.storylyInit = StorylyInit(storylyId: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40")
        self.storylyView.rootViewController = self
        
        let storylyViewProgrammatic = StorylyView()
        storylyViewProgrammatic.translatesAutoresizingMaskIntoConstraints = false
        storylyViewProgrammatic.storylyInit = StorylyInit(storylyId: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40")
        storylyViewProgrammatic.rootViewController = self
        self.view.addSubview(storylyViewProgrammatic)
        storylyViewProgrammatic.leadingAnchor.constraint(equalTo: self.view.leadingAnchor).isActive = true
        storylyViewProgrammatic.trailingAnchor.constraint(equalTo: self.view.trailingAnchor).isActive = true
        storylyViewProgrammatic.topAnchor.constraint(equalTo: self.storylyView.bottomAnchor, constant: 10).isActive = true
        storylyViewProgrammatic.heightAnchor.constraint(equalToConstant: 120).isActive = true
    }
}
